odoo/addons/base/models/ir_mail_server.py

```diff
@@ -153,14 +153,14 @@ def _compute_smtp_authentication_info(self):
             else:
                 server.smtp_authentication = False
 
-    @api.constrains('smtp_ssl_certificate', 'smtp_ssl_private_key')
+    @api.constrains('smtp_authentication', 'smtp_ssl_certificate', 'smtp_ssl_private_key')
     def _check_smtp_ssl_files(self):
-        """We must provided both files or none."""
         for mail_server in self:
-            if mail_server.smtp_ssl_certificate and not mail_server.smtp_ssl_private_key:
-                raise UserError(_('SSL private key is missing for %s.', mail_server.name))
-            elif mail_server.smtp_ssl_private_key and not mail_server.smtp_ssl_certificate:
-                raise UserError(_('SSL certificate is missing for %s.', mail_server.name))
+            if mail_server.smtp_authentication == 'certificate':
```

Comment:
I did think about it but I don't think it is a good idea.

Let's take a noob user who doesn't understand shit about oauth/certificates/tls. He is struggling on the ir.mail_server form view. He is changing the `smtp_authentication` back and forth between `login` and `certificate` trying to understand what is going on.

At the moment he can leave the two `smtp_ssl_private_key` and `smtp_ssl_certificate` dangling. In case he tries with `smtp_authentication='login'` those two fields are simply ignored. If he goes back to `smtp_authentication='certificate'` the two fields just continue to hold their previous value, he doesn't have to supply them again.

With the change you suggest, an UserError would show up asking him to remove those two files... but since he is on `smtp_authentication != 'certificate'` neither field would show up on the form view. He would need first to select `smtp_authentication='certificate'` to be able to remove the two fields and then go back on `login` to test his shit. In case he changes is mind again and come back on `smtp_authentication='certificate'` then he would have to supply both fields again...

I don't think it is a good idea :D

Reply:
or why not keeping the old condition then, when `mail_server.smtp_authentication != 'certificate'` ? (both should be set or none)